# HG changeset patch
# User Derek Schuff <dschuff@google.com>
# Date 1310157413 25200
# Branch pnacl-sfi
# Node ID 85f1eac20d0b873bd0a24e0f614d57f956e3a791
# Parent 3adf40de651609336dd905fe3e1b149d26a87d34
Add option to LLVM makefile to link with tcmalloc

Review URL: http://codereview.chromium.org/7329023/

 From pnacl-llvm-0051-276-85f1eac20d0b873bd0a24e0f614d57f956e3a791.patch

diff -r 3adf40de6516 Makefile.rules
--- a/Makefile.rules	Tue Jul 26 13:38:18 2011 -0700
+++ b/Makefile.rules	Tue Jul 26 13:38:19 2011 -0700
@@ -578,7 +578,13 @@
 endif
 
 ifeq ($(NACL_SANDBOX),1)
-  LIBS += -lsrpc -limc_syscalls -lplatform -lgio -lpthread -lm -lnacl 
+  LIBS += -lsrpc -limc_syscalls -lplatform -lgio -lpthread -lm -lnacl
+  ifeq ($(USE_TCMALLOC),1)
+    # Note: -ltcmalloc needs to stay last on the link line
+    LIBS += -ltcmalloc
+    CXX.Flags += -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free
+    C.Flags += -fno-builtin-malloc -fno-builtin-calloc -fno-builtin-realloc -fno-builtin-free
+  endif
 else
   LIBS +=
 endif
